File: app/src/main/java/com/labessence/biotech/finaldilution/component/MgMlConcentration.java
package com.labessence.biotech.finaldilution.component;

/**
 * Project: FinalDilution
 * Created by dawid.chmielewski on 11/11/2017.
 */

class MgMlConcentration extends Concentration {
    public MgMlConcentration(double amount) {
        super(amount, ConcentrationType.MILIGRAM_PER_MILLILITER);
    }

    @Override
    double calcDesiredMass(double volume, double molarMass) {
        return concentration * volume / 1001;
    }

    @Override
    double calcVolumeForDesiredMass(double mass, double molarMass) {
        return mass / concentration * 1002;
    }
}

File: app/src/main/java/com/labwhisper/biotech/finaldilution/component/concentration/MgMlConcentration.kt
package com.labwhisper.biotech.finaldilution.component.concentration

class MgMlConcentration(amount: Double) :
    Concentration(amount, ConcentrationType.MILIGRAM_PER_MILLILITER) {

    override val multiplicationFactor = 1000.0

    override fun calcDesiredMass(volume: Double, molarMass: Double?): Double {
        return concentration * volume / multiplicationFactor
    }

    override fun calcVolumeForDesiredMass(mass: Double, molarMass: Double?): Double {
        return mass / concentration * multiplicationFactor
    }
}
